src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -94,70 +95,111 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
+	private _registerCommandHandlers(): void {
 		const terminal = this._terminal;
 		const commandDetection = this._capabilities.get(TerminalCapability.CommandDetection);
 		if (!terminal || !commandDetection) {
 			return;
 		}
-		this._register(commandDetection.onCommandExecuted(() => {
-			this._decoration?.dispose();
-			this._decoration = undefined;
-		}));
-		this._register(commandDetection.onCommandFinished(async command => {
-			this._decoration?.dispose();
-			this._decoration = undefined;
-			this._quickFixes = getQuickFixes(command, this._commandListeners, this._openerService, this._onDidRequestRerunCommand);
-		}));
+		this._register(commandDetection.onCommandFinished(command => this._resolveQuickFixes(command)));
 		// The buffer is not ready by the time command finish
-		// is called. Add the decoration on command start using the actions, if any,
-		// from the last command
+		// is called. Add the decoration on command start if there are corresponding quick fixes
 		this._register(commandDetection.onCommandStarted(() => {
-			if (this._quickFixes) {
-				this._registerContextualDecoration();
-				this._quickFixes = undefined;
+			const registeredDecoration = this._registerQuickFixDecoration();
+			if (registeredDecoration && this._quickFixId) {
+				this._quickFixId++;
 			}
 		}));
 	}
 
-	private _registerContextualDecoration(): void {
-		if (!this._terminal) {
+	/**
+	 * Resolves quick fixes, if any, based on the
+	 * @param command & its output
+	 * and establishes an event which will update the quick fixes
+	 * when one has been run
+	 */
+	private _resolveQuickFixes(command: ITerminalCommand): void {
+		if (command.command !== '') {
+			this._disposeQuickFix();
+		} else if (this._currentQuickFix && this._quickFixId) {
+			// move it to the bottom
+			this._quickFixId--;
+			if (this._quickFixId < 1) {
+				this._quickFixId = 1;
+			}
+			const decoration = this._decorations.get(this._quickFixId);
+			decoration?.dispose();
+			this._decorations.delete(this._quickFixId);
+			this._registerQuickFixDecoration();
 			return;
 		}
+		const result = getQuickFixesForCommand(command, this._commandListeners, this._openerService, this._onDidRequestRerunCommand);
+		if (!result) {
+			return;
+		}
+		const { fixes, onDidRunQuickFix } = result;
+		const id = this._quickFixId || 1;
+		this._quickFixId = id;
+		this._quickFixes.set(id, fixes);
+		onDidRunQuickFix(() => this._disposeQuickFix());
+	}
+
+	private _disposeQuickFix(): void {
+		this._currentQuickFix?.dispose();
+		if (this._quickFixId) {
+			this._decorations.delete(this._quickFixId);
+			this._quickFixes.delete(this._quickFixId);
+		}
+	}
+
+	/**
+	 * Registers a decoration with the quick fixes
+	 */
+	private _registerQuickFixDecoration(): boolean {
+		if (!this._terminal) {
+			return false;
+		}
+		const id = this._quickFixId;
+		if (!id || !this._quickFixes.get(id) || this._decorations.get(id)) {
+			return false;
+		}
 		const marker = this._terminal.registerMarker();
```

Comment:
The decoration may not be positioned correctly after enter:

![image](https://user-images.githubusercontent.com/2193314/194335950-098f22e3-ece7-45e9-9e98-7097746f0e60.png)

Might only happen on Windows after the repositioning of the decoration that can occur.

Reply:
I'm not convinced that we should be repositioning the decoration at all. It adds complexity like this and doesn't add much benefit considering the shortcut will still work as will clicking on it several lines above

Reply:
That's fair. In that case, could we still keep it on top of the previous placeholder? It moves atm:

![Recording 2022-10-06 at 07 28 55](https://user-images.githubusercontent.com/2193314/194339900-09255a48-ef79-4f4e-9fc8-bdc2892b7c57.gif)

Reply:
I'll just undo the changes that make it move